### ISSUE TITLE
The location of the icons on the top bar [affected by the position weight] get's set incorrectly between GNOME shell startups 

### ISSUE THREAD

**Issue by linuxiorr** [martinpl/tray-icons-reloaded#7]
I can set the desired position weight, in my case it's position weight 6 to make it appear as pictured in the first screenshot.

When I restart my system, on system startup the position of the icons on the top bar gets set incorrectly (as pictured on the second screenshot, they appear on the right-most position on the panel). To make it appear as desired again, I have to set the position weight to a different value first, and then set it back to the desired value to make it appear correct again.

The same problem also happens when I use ALT+F2 to restart the shell.
![tray-icons-report](https://user-images.githubusercontent.com/33715018/79979393-bcc74b00-84a1-11ea-8a90-685530023ee0.png)
![tray-icons-report-2](https://user-images.githubusercontent.com/33715018/79979431-cbadfd80-84a1-11ea-8f71-bf0da96072f2.png)

**Comment by martinpl** [comment]
Delayed plugin startup should fix this issue, try this: (rename to extension.js and drop to ~/.local/share/gnome-shell/extensions/trayIconsReloaded@selfmade.pl)
[extension.js.txt](https://github.com/MartinPL/Tray-Icons-Reloaded/files/4517297/extension.js.txt)

**Comment by linuxiorr** [comment]
After I replaced the original extension.js file with the new one, the extension stopped working completely (I've tried it with both the version from the GNOME extensions website, as well as the cloned git version).

The icons don't appear on the top, and the extension is greyed out in the GNOME extensions app (it's listed and I can open the configuration window but it can't be enabled).

**Comment by martinpl** [comment]
I forgot add GLib 😅, just add `const GLib = imports.gi.GLib;` on top. File is for extensions.gnome.org release.

**Comment by linuxiorr** [comment]
I added the line at the top.

It partially fixed the issue.

Now after reboots, the icons don't appear on the right most position. Instead, they appear right before the GNOME grouped drop down menu as pictured on the screenshot (as if the weight had been set to 7).
Setting the weight to 4-6 makes it appear as weight 7 after the restart.

![tray-afrer-restart](https://user-images.githubusercontent.com/33715018/80099334-93282580-856e-11ea-8bbc-c6b42074db52.png)


On the other hand if I set the weight to 7, after I restart the shell the icons start appearing at the end of the top bar again.

![tray-after-restart-vol](https://user-images.githubusercontent.com/33715018/80100757-af2cc680-8570-11ea-8493-df03fa250fc3.png)


If I set the weight to 3, and restart the shell, after the restart it appears on my preferred position as If I had set it to my desired value of 6.

![tray-vol-3](https://user-images.githubusercontent.com/33715018/80102425-ff0c8d00-8572-11ea-9dd3-566077d30867.png)

**Comment by martinpl** [comment]
Try modify timer from 500 to 1000 or even more. (same file, search it by CTRL + F)

**Comment by martinpl** [comment]
If others extension indicator appear after tray icons, that's with probably will not be easy fixable

**Comment by martinpl** [comment]
Any luck with current version in extensions.gnome.org or master?

**Comment by linuxiorr** [comment]
The behaviour is still the same with the current version from extensions.gnome.org

I've resorted to just setting the weight to 3, as the crucial thing to me, is that after I start my system / restart the shell, the icons are in my preferred position.

This behaviour might prove tricky for some users though, If they don't know that the weight option doesn't work in a fully straightforward way (the position on the top bar after starting / restarting the shell is different than the position that's displayed after setting the weight in the configuration options).

Also, I had tried modifying the timer with a previous version - it didn't fix the issue.
